### PR TITLE
cleanup of EntityItem ctors and friends

### PR DESCRIPTION
**Pull request by AndrewMeadows**
Philip noticed a bug where sphere entities were being created via script and would momentarily show up with DEFAULT dimensions before their correct size.  While trying to look into it I noticed some cruft in the Entity constructors and decided to clean them up.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 18d0cb5158f7d5ebd140060f25f6d82ffef1cec0 available at: https://gist.github.com/8b30b339c553ae070673<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.dmg

File: libraries/entities/src/BoxEntityItem.cpp [hunk 1]
```diff
@@ -29,7 +29,6 @@ BoxEntityItem::BoxEntityItem(const EntityItemID& entityItemID, const EntityItemP
         EntityItem(entityItemID) 
 {
     _type = EntityTypes::Box;
-    _created = properties.getCreated();
     setProperties(properties);
 }
```
Comment (AndrewMeadows) [inline libraries/entities/src/BoxEntityItem.cpp]: We can speculate as to why **_created** was being siphoned out of properties before the **setProperties()** call here.  My main theory was that we once were trying to make sure we always got a valid **_created** time in the EntityItem constructors (note that **_created** wouldn't be copied from the **properties** unless it had been explicitly marked as **changed**).  However, this is now being enforced in **EntityItem::setProperties()** (if both _created and _timestamp are init values) or via **EntityItem::recordCreationTime()** in EntityTree::addEntity() (if an entity is actually added).  Hence, I believe this call here (and its friends in other ctors) is cruft.

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -85,11 +85,6 @@ EntityItem::EntityItem(const EntityItemID& entityItemID) :
     _lastUpdated = now;
 }
 
-EntityItem::EntityItem(const EntityItemID& entityItemID, const EntityItemProperties& properties) : EntityItem(entityItemID)
-{
-    setProperties(properties);
-}
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityItem.cpp]: This ctor did not properly initialize data members, and isn't needed anyway -- we can use the default ctor and then explicitly call **setProperties()** in the derived class ctors.

File: libraries/entities/src/PolyVoxEntityItem.cpp [hunk 3]
```diff
@@ -57,7 +57,6 @@ PolyVoxEntityItem::PolyVoxEntityItem(const EntityItemID& entityItemID, const Ent
     _voxelSurfaceStyle(PolyVoxEntityItem::DEFAULT_VOXEL_SURFACE_STYLE)
 {
     _type = EntityTypes::PolyVox;
-    _created = properties.getCreated();
```
Comment (ZappoMan) [inline libraries/entities/src/PolyVoxEntityItem.cpp]: did you test that loading a file from json still results in proper create times and age?

**Comment by PhilipRosedale** [comment]
Does this fix the size problem?  Note that the incorrect size persisted (it was not momentary as you mentioned in the writeup).

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2441be21e29175518c41e056690cfad3131ec0aa available at: https://gist.github.com/8b30b339c553ae070673<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.dmg

File: libraries/entities/src/PolyVoxEntityItem.cpp (hunk 3 above)
Comment (AndrewMeadows) [inline libraries/entities/src/PolyVoxEntityItem.cpp]: No.  I did not test.

File: libraries/entities-renderer/src/RenderableSphereEntityItem.cpp [hunk 4]
```diff
@@ -39,7 +39,15 @@ void RenderableSphereEntityItem::render(RenderArgs* args) {
...
+#define USE_MAGIC_WORKAROUND
+#ifdef USE_MAGIC_WORKAROUND
+    // It's a mystery how this actually works!
+    Transform transform = getTransformToCenter();
+    transform.setScale(transform.getScale());
+    batch.setModelTransform(transform); // use a transform with scale, rotation, registration point and translation
+#else // USE_MAGIC_WORKAROUND
     batch.setModelTransform(getTransformToCenter()); // use a transform with scale, rotation, registration point and translation
+#endif // USE_MAGIC_WORKAROUND
     DependencyManager::get<DeferredLightingEffect>()->renderSolidSphere(batch, 0.5f, SLICES, STACKS, sphereColor);
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableSphereEntityItem.cpp]: This is a more correct fix for bad sphere radius (when sphere dimensions are specified as <0.1, 0.1, 0.1>).  Why it works is a mystery but Sam might have an idea.

File: libraries/entities/src/PolyVoxEntityItem.cpp (hunk 3 above)
Comment (AndrewMeadows) [inline libraries/entities/src/PolyVoxEntityItem.cpp]: This is tested: creation time of polyvox entities do indeed survive saving to JSON and back again.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1c7afbda04d2ff4032a6ffb73133c44c8be9bc5d available at: https://gist.github.com/8b30b339c553ae070673<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.dmg

**Comment by howard-stearns** [comment]
I don't think this fixes the problem. Try the script at http://howard-stearns.github.io/models/scripts/entitySize.js

It produces a sphere and a cube at your avatar position, both of size 0.1.
The dimensions report correctly, but, on my machine, the cube displays much bigger. 
(You can tell by creating another cube interactively with edit.js, and moving it next to the one created by the script. Or you can edit say, the box and increase it's y position and compare it to the sphere.)

Sometimes quitting the script and then Interface, and coming back to the persisted space fixes the problem, and sometimes it doesn't. In previous builds, it was the sphere that was rendering big, and as far as I know, was always fixed before by quitting and re-entering.

**Comment by AndrewMeadows** [comment]
Ok, this time its really fixed for all entity types.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 32b1fe0918f4755d5f5d6720d908f0bab0232029 available at: https://gist.github.com/8b30b339c553ae070673<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5353.dmg

**Comment by howard-stearns** [comment]
Ah, very good. Works For Me.

File: libraries/entities/src/EntityTreeElement.cpp [hunk 5]
```diff
@@ -772,6 +772,9 @@ int EntityTreeElement::readElementDataFromBuffer(const unsigned char* data, int
                         entityItemID = entityItem->getEntityItemID();
                         _myTree->setContainingElement(entityItemID, this);
                         _myTree->postAddEntity(entityItem);
+                        if (entityItem->getCreated() == UNKNOWN_CREATED_TIME) {
+                            entityItem->recordCreationTime();
```
Comment (ZappoMan) [inline libraries/entities/src/EntityTreeElement.cpp]: we probably want to do this before we add it to the tree...

Comment (AndrewMeadows) [inline libraries/entities/src/EntityTreeElement.cpp]: We already do it: in EntityTree::addEntity().

**Comment by howard-stearns** [comment]
Cleaner, too.

**Pull request merged**